### ISSUE TITLE
Validating onfocusout usability concerns

### ISSUE THREAD

**Issue by biljanap** [wet-boew/wet-boew#8353]
This has been already discussed here  #8160 but from WCAG compliance perspective.
This behavior seems to be causing a lot of issues and confusion for many people.

Automatically modifying the label to include error messages is causing the page to shift vertically. It’s very annoying and a lot of people complain that they ended up completing the wrong fields as a result…which in return created more validation errors. 
Also we received many comments that this behavior is distracting and people lose the focus especially on longer forms. 
We believe this became a usability issue and we wonder if the negatives outweigh positives.

We received the same comments from different people involved in our latest project (testers, clients, developers).
Testers and clients raised it as an issue and filed it as a bug multiple times. We spent a lot of times explaining and justifying the feature but it keeps coming back.

I also think that heading text “The form could not be submitted because (#) error was found.” is misleading and confusing. The text makes sense only if the user is trying to submit a form with invalid or missing mandatory fields. The text is misleading if the user did not click on Submit yet. It could be argued that this is usability issue as well.

I don’t mean to challenge the validity of this feature but I am wondering if this automatic validating onfocusout could be made optional somehow.

**Comment by RobJohnston** [comment]
@biljanap : The heading text was also brought up in https://github.com/wet-boew/wet-boew/issues/7394.

**Comment by duboisp** [comment]
My recommendation would be to define a usability testing plan.
Creating a series of test case with actual working example where it would also explain how to reproduce the issue, explain what is the expected result and explain what is current result.
And then it will be a matter to perform a usability research in order to validate or invalidate the previously expectation and then reporting the findings along with all the UX research data.

**Comment by BenTrem** [comment]
@duboisp - I think you've drilled into a rich vein here. And so am moved to suggest: this ain't the platform that would support discussion of such rich material.
I've been online since 1972 (arguably). WWW since 1994. 2018 and I see tsunami of options and desert of effective thinking.

When it comes to effective tactic (UX and such as that) we can refer to time-proven resources. Can ... could ...

I've blown decades preaching best practice so here essay hoping that someone sees the value of pursuing these matters in depth.

**Comment by BenTrem** [comment]
@duboisp - I think you've drilled into a rich vein here. And so am moved to suggest: this ain't the platform that would support discussion of such rich material.
I've been online since 1972 (arguably). WWW since 1994. 2018 and I see tsunami of options and desert of effective thinking.

When it comes to effective tactic (UX and such as that) we can refer to time-proven resources. Can ... could ...

I've blown decades preaching best practice so here essay hoping that someone sees the value of pursuing these matters in depth.

p.s. was chatting with a long-lived friend about his most recent presentation.
"Many organizations weigh the benefits and risks of data lakes versus data warehouses. The former can be vast but slow, while the latter require great effort to achieve desired speed. But these solutions are no longer mutually exclusive."
Have a look at the URL he presents so we can share the resource:
"https://bloorgroup.webex.com/mw3200/mywebex/default.do?nomenu=true&siteurl=bloorgroup&service=6&rnd=0.5141840059127001&main_url=https%3A%2F%2Fbloorgroup.webex.com%2Fec3200%2Feventcenter%2Fevent%2FeventAction.do%3FtheAction%3Ddetail%26%26%26EMK%3D4832534b000000043e9707f6a17d52da187cd9c155faa298d37fa021f86debd4ddabebb6cfbca075%26siteurl%3Dbloorgroup%26confViewID%3D92488319211962181%26SourceId%3Dcc%26encryptTicket%3DSDJTSwAAAARkcOENtOwJWpzGESXavUzb2fBnz0Ugd-AJ65WAzspVxQ2%26"

**Comment by duboisp** [comment]
@BenTrem  
> I think you've drilled into a rich vein here.

I recently created the [WET project documentation](https://wet-boew.github.io/wet-boew-documentation/index-en.html) website which would contain that kind of information. Having that information centralized would help the project maintenance and provide the rational behind some design pattern that we are using.

**Comment by BenTrem** [comment]
@duboisp Thanks for that; I'll spider it "real soon". But what I was addressing was the ability (or lack of same) to ... what ... spit-ball. Or blue-sky.
50yrs ago I would sit back and kick ideas around via ham radio RTTY. 20+ years ago, that productive activity via IRC. Perhaps we are slipping into the dark ages because people don't feel the need for even branched comments.

In my experience when team leaders leave the room the guys who actually do the lifting toss ideas around.

Apologies for the essay.
